File: week-3/lemonade-change/lemonade-change.py
class Solution:
    def lemonadeChange(self, bills: List[int]) -> bool:
        fives = 0
        tens = 0
        
        for i in bills:
            if i == 10:
                if fives > 0:
                    fives -= 1
                    tens += 1
                else:
                    return False
            elif i == 5:
                fives += 1
            else:
                if fives > 0 and tens > 0:
                    fives -= 1
                    tens -= 1
                elif fives >= 3:
                    fives -= 3
                else:
                    return False
            
        return True

File: week-3/lemonade-change/lemonade-change.js
/**
 * @param {number[]} bills
 * @return {boolean}
 */
var lemonadeChange = function(bills) {
    let fives = 0, tens = 0;
    
    for (let i of bills) {
        if (i === 5) ++fives;
        else if (i === 10)
            if (fives > 0) {
                --fives; ++tens;
            } else return false
        else {
            if (fives>0 && tens>0) {
                --fives;
                --tens;
            } else if (fives >= 3) {
                fives -= 3;
            } else return false;
        }
    }
    
    return true
};
